### PR TITLE
packaging: updating packaging script to support Buck binary output

### PR DESCRIPTION
**Pull request by muffins**
## Summary:

With the introduction and support of the Buck build system, we should start adding support to our various build|provision|deployment scripts for the Buck components. This adds in checks for the buck binaries as a secondary option to the CMake binary output to make generation of osquery packages easy for either build platform.

### CMake:

- Build the binaries:

```
PS C:\Users\Nicholas\work\repos\osquery> cmake --build .\build\windows10\ --config Release -j10                                                                                                     Microsoft (R) Build Engine version 16.1.76+g14b0a930a7 for .NET Framework
Copyright (C) Microsoft Corporation. All rights reserved.

  osquery_utils_config.vcxproj -> C:\Users\Nicholas\work\repos\osquery\build\windows10\osquery\utils\config\Release\osquery_utils_config.lib
  osquery_utils_system_env.vcxproj -> C:\Users\Nicholas\work\repos\osquery\build\windows10\osquery\utils\system\Release\osquery_utils_system_env.lib
  osquery_utils_system_time.vcxproj -> C:\Users\Nicholas\work\repos\osquery\build\windows10\osquery\utils\system\Release\osquery_utils_system_time.lib
  osquery_utils_system_errno.vcxproj -> C:\Users\Nicholas\work\repos\osquery\build\windows10\osquery\utils\system\Release\osquery_utils_system_errno.lib
  osquery_utils_status.vcxproj -> C:\Users\Nicholas\work\repos\osquery\build\windows10\osquery\utils\status\Release\osquery_utils_status.lib
  Initializing custom PYTHONPATH: C:/Users/Nicholas/work/repos/osquery/build/windows10/python_path
...
```

- Package the binaries:
```
PS C:\Users\Nicholas\work\repos\osquery> .\tools\deployment\make_windows_package.ps1
[*] Building osquery Chocolatey package
[*] Did not find openssl certs.pem
[+] Chocolatey Package written to C:\Users\Nicholas\work\repos\osquery\build\chocolatey\osquery\osquery.4.0.0.nupkg 
```

- Install the binaries:
```
PS C:\Users\Nicholas\Desktop> clist -lair | sls osquery
PS C:\Users\Nicholas\Desktop> choco install -yf -s . osquery --params='/InstallService' --version 4.0.0
Chocolatey v0.10.15
Installing the following packages:
osquery
By installing you accept licenses for the packages.

osquery v4.0.0 (forced)
osquery package files install completed. Performing other installation steps.
C:\Program Files\osquery\osqueryd
C:\Program Files\osquery\log
Extracting C:\ProgramData\chocolatey\lib\osquery\tools\\bin\\osquery.zip to C:\Program Files\osquery...
C:\Program Files\osquery
True
osqueryd
Environment Vars (like PATH) have changed. Close/reopen your shell to
 see the changes (or in powershell/cmd.exe just type `refreshenv`).
 The install of osquery was successful.
  Software installed to 'C:\Program Files\osquery'

Chocolatey installed 1/1 packages.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).
PS C:\Users\Nicholas\Desktop> sc.exe qc osqueryd                                                                                  [SC] QueryServiceConfig SUCCESS

SERVICE_NAME: osqueryd
        TYPE               : 10  WIN32_OWN_PROCESS
        START_TYPE         : 2   AUTO_START
        ERROR_CONTROL      : 1   NORMAL
        BINARY_PATH_NAME   : "C:\Program Files\osquery\osqueryd\osqueryd.exe" --flagfile="C:\Program Files\osquery\osquery.flags"
        LOAD_ORDER_GROUP   :
        TAG                : 0
        DISPLAY_NAME       : osqueryd
        DEPENDENCIES       :
        SERVICE_START_NAME : LocalSystem
PS C:\Users\Nicholas\Desktop>
```

### Buck:

- Build the binaries:
```
PS C:\Users\Nicholas\work\repos\osquery> buck build @mode/windows-x86_64/release osquery:osqueryd                                                      Not using buckd because watchman isn't installed.
Parsing buck files: finished in 2.6 sec (100%)
Creating action graph: finished in 0.7 sec (100%)
Building: finished in 2.6 sec (100%) 113/113 jobs, 0 updated
  Total time: 6.0 sec
```

- Package the binaries:
```
PS C:\Users\Nicholas\work\repos\osquery> .\tools\deployment\make_windows_package.ps1 -Build 'buck'
[*] Building osquery Chocolatey package
[*] Did not find openssl certs.pem
[+] Chocolatey Package written to C:\Users\Nicholas\work\repos\osquery\build\chocolatey\osquery\osquery.4.0.0.nupkg
```

- Install the binaries:
```
PS C:\Users\Nicholas\Desktop> choco install -yf -s . osquery --params='/InstallService' --version 4.0.0
Chocolatey v0.10.15
Installing the following packages:
osquery
By installing you accept licenses for the packages.

osquery v4.0.0 (forced)
osquery package files install completed. Performing other installation steps.
C:\Program Files\osquery\osqueryd
C:\Program Files\osquery\log
Extracting C:\ProgramData\chocolatey\lib\osquery\tools\\bin\\osquery.zip to C:\Program Files\osquery...
C:\Program Files\osquery
True
osqueryd
Environment Vars (like PATH) have changed. Close/reopen your shell to
 see the changes (or in powershell/cmd.exe just type `refreshenv`).
 The install of osquery was successful.
  Software installed to 'C:\Program Files\osquery'

Chocolatey installed 1/1 packages.
 See the log for details (C:\ProgramData\chocolatey\logs\chocolatey.log).

Enjoy using Chocolatey? Explore more amazing features to take your
experience to the next level at
 https://chocolatey.org/compare
PS C:\Users\Nicholas\Desktop> sc.exe qc osqueryd                                                                                                       [SC] QueryServiceConfig SUCCESS

SERVICE_NAME: osqueryd
        TYPE               : 10  WIN32_OWN_PROCESS
        START_TYPE         : 2   AUTO_START
        ERROR_CONTROL      : 1   NORMAL
        BINARY_PATH_NAME   : "C:\Program Files\osquery\osqueryd\osqueryd.exe" --flagfile="C:\Program Files\osquery\osquery.flags"
        LOAD_ORDER_GROUP   :
        TAG                : 0
        DISPLAY_NAME       : osqueryd
        DEPENDENCIES       :
        SERVICE_START_NAME : LocalSystem
        
PS C:\Users\Nicholas\Desktop> get-service osqueryd                                                                                                     
Status   Name               DisplayName
------   ----               -----------
Running  osqueryd           osqueryd
```

### CODE REVIEW

File: tools/deployment/make_windows_package.ps1 [hunk 1]
```diff
@@ -605,17 +606,30 @@ function Main() {
...
 
-  $scriptPath = Get-Location
-  $buildPath = Join-Path $scriptPath 'build\windows10\osquery\Release'
-  $daemon = Join-Path $buildPath 'osqueryd.exe'
-  $shell = Join-Path $buildPath 'osqueryi.exe'
+  $buildPath = ''
+  if ($Build -eq 'buck') {
+    $buildPath = Join-Path $osqRoot 'buck-out\release\gen\osquery'
+  } else {
+    $buildPath = Join-Path $osqRoot 'build\windows10\osquery\Release'
```
Comment (theopolis) [inline tools/deployment/make_windows_package.ps1]: This makes an assumption about where `cmake ..` was run. Is there a more flexible way to discover this path?

Comment (theopolis) [reply tools/deployment/make_windows_package.ps1]: Perhaps instead of taking in `BuildType` it can take `buildPath` directly?

**Comment by theopolis** [comment]
Any updates on this, should we merge?

File: tools/deployment/make_windows_package.ps1 (hunk 1 above)
Comment (muffins) [reply tools/deployment/make_windows_package.ps1]: Sure, I'll get this fixed up tonight, sorry for the delay!

**Comment by muffins** [comment]
@theopolis I respond inline, I'll get this fixed up before landing. I was trying to stick to defaults we had used in the past for the build directories, but yeah thinking about it I'm not really sure that's absolutely necessary, considering some of the build scripts don't really even work anymore.

**Comment by theopolis** [comment]
Thanks @muffins!

**Pull request merged**